src/grpc_interceptor/client.py

```diff
@@ -0,0 +1,172 @@
+"""Base class for client-side interceptors."""
+
+import abc
+from collections import namedtuple
+from typing import Any, Callable, Iterator, Optional, Tuple
+
+import grpc
+
+
+class ClientCallDetails(
```

Comment:
I would prefer to just use `grpc.ClientCallDetails` and not wrap / redefine it. Is there a reason that doesn't work?

Reply:
`grpc.ClientCallDetails` doesn't actually declare any of its attributes. That really only comes up in testing though now that you mention it. I can move this out to testing and just use `grpc.ClientCallDetails` itself here. Do you have any preference between putting it in `dummy_client` vs making a new module for data containers like this?

Reply:
I went back and forth on this. Ended up leaving the `ClientCallDetails` wrapper as a convenience class and changing the `ClientInterceptor` param types to the base `grpc.ClientCallDetails`.

The `ClientCallDetails` instance passed into the interceptor is immutable so a new instance needs to be built if you want to modify anything. Since `grpc.ClientCallDetails` itself does not declare the attributes you need to either just "know" what attributes to copy over or refer back to the module documentation. Having this parameterized wrapper eases that but is totally optional.